src/treq/response.py

```diff
@@ -95,3 +97,25 @@ def cookies(self):
                 jar.set_cookie(cookie)
 
         return jar
+
+    def fail_for_status(self):
+        """
+        Fail with an error if the response has an error status.
+
+        :rtype: A `Deferred` that fires with the response itself if the status
+            is a known non-error code, or that calls its errback chain with a
+            :class:`twisted.web.error.Error` otherwise.
+        """
+        if _is_error_code(self.code):
+            d = defer.succeed(self)
+        else:
+            d = self.content().addCallback(self._raise_error)
```

Comment:
yes, very confusing, thanks for the catch. I actually just removed the helper since it's one line and only used once.